File: backend/schemas/ai.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID
from decimal import Decimal


class AIAnalysisRequest(BaseModel):
    months: int = 6


class BudgetRecommendation(BaseModel):
    category_id: UUID
    category_name: str
    recommended_limit: Decimal
    reasoning: str


class AIAnalysisResponse(BaseModel):
    recommendations: list[BudgetRecommendation]
    patterns: list[str]
    suggestions: list[str]


class LifeEventRequest(BaseModel):
    event_type: str
    event_date: str  # ISO date string
    description: Optional[str] = None


class LifeEventResponse(BaseModel):
    id: UUID
    event_type: str
    event_date: str
    description: Optional[str]
    created_at: datetime
    
    class Config:
        from_attributes = True


class AIInsightResponse(BaseModel):
    id: UUID
    insight_type: Optional[str]
    content: str
    created_at: datetime
    is_read: bool
    
    class Config:
        from_attributes = True


class AIAskRequest(BaseModel):
    question: str


class AIAskResponse(BaseModel):
    answer: str

File: frontend/src/services/ai.ts
import api from './api';

export interface AIAnalysisRequest {
  months: number;
}

export interface AIAnalysisResponse {
  recommendations: Array<{
    category_id?: string;
    category_name: string;
    recommended_limit: number;
    reasoning: string;
  }>;
  patterns: string[];
  suggestions: string[];
}

export interface LifeEventRequest {
  event_type: string;
  event_date: string;
  description?: string;
}

export interface AIInsight {
  id: string;
  insight_type?: string;
  content: string;
  created_at: string;
  is_read: boolean;
}

export interface AIAskRequest {
  question: string;
}

export interface AIAskResponse {
  answer: string;
}

export const aiService = {
  analyze: async (months: number = 6): Promise<AIAnalysisResponse> => {
    const response = await api.post('/api/ai/analyze', { months });
    return response.data;
  },

  logLifeEvent: async (data: LifeEventRequest) => {
    const response = await api.post('/api/ai/life-event', data);
    return response.data;
  },

  getInsights: async (): Promise<AIInsight[]> => {
    const response = await api.get('/api/ai/insights');
    return response.data;
  },

  ask: async (question: string): Promise<string> => {
    const response = await api.post('/api/ai/ask', { question });
    return response.data.answer;
  },
};

